app/assets/javascripts/discourse/app/components/modal/create-account.js

```diff
@@ -507,17 +474,14 @@ export default class CreateAccount extends Component.extend(
     );
   }
 
-  @discourseComputed(
-    "model.authOptions.associate_url",
-    "model.authOptions.auth_provider"
-  )
-  associateHtml(url, provider) {
+  get associateHtml() {
+    const url = this.authOptions?.associate_url;
     if (!url) {
       return;
     }
     return i18n("create_account.associate", {
       associate_link: url,
-      provider: i18n(`login.${provider}.name`),
+      provider: i18n(`login.${this.authOptions?.auth_provider}.name`),
```

Comment:
```suggestion
      provider: i18n(`login.${this.authOptions.auth_provider}.name`),
```

This code path will never get invoked if `this.authOptions === null`, right?

Reply:
`associateHtml` gets called when `this.showCreateForm` evaluates to true:
https://github.com/discourse/discourse/blob/703154f26b5f7f9b6b8eebb0293ebb727e95e04a/app/assets/javascripts/discourse/app/components/modal/create-account.hbs#L34-L40

And `this.showCreateForm` can be true even if this.authOptions is null and `this.canCreateLocal`(a setting) is true:
https://github.com/discourse/discourse/blob/7dd52eb112c02d691c06a465a2a8550c93a40b0a/app/assets/javascripts/discourse/app/components/modal/create-account.js#L121-L125

Reply:
Ok. This is is all minor, but used as a translation string, the `?.` operator is odd, because when it is empty, it would result in `login.undefined.name` here, and we won't have a locale entry for that, right? 

Also, just above, on the `const url = this.authOptions?.associate_url; if (!url) { return; }` lines, we're indirectly checking for the presence of `this.authpOtions, if that is empty, this bit is skipped.

Reply:
Ah yes, that does make sense, I missed that implication of the above conditional 👍